### PR TITLE
Go version

### PR DESCRIPTION
**Pull request by martinsson**
Not much to say :)

I can think of some improvements but I think this is a good enough version of it.

Cheers
Johan

### CODE REVIEW

**Review by tomphp — COMMENTED** [review]
This is great, thank you!

I'm happy to merge this now, but I have a couple of suggestions about idiomatic Go projects.

1. Go projects typically use lowercase snake_case filenames.
2. Go has a very opinionated, official style guide. If you run `go fmt` in the `src` folder it will reformat automatically.

File: go/src/Pipeline_test.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+	pipeline.run(project)
+
+	assert.Equal(t, []string{
+		"INFO: Tests passed",
+		"ERROR: Deployment failed",
+		"INFO: Email disabled"}, logger.getLoggedLines())
+	assert.False(t, emailer.wascalled)
+}
+
+func TestProject_withou_tests_that_fails_build_with_email_notification(t *testing.T) {
```
Comment (tomphp) [inline go/src/Pipeline_test.go]: Also, these tests have a typo (missing t on `without`) ;)

**Comment by martinsson** [comment]
Thanks for the feedback.

Done

**Comment by tomphp** [comment]
Thanks for submitting this, it's really awesome!

**Comment by martinsson** [comment]
My pleasure. Interesting kata by the way.

**Pull request merged**